cmd/dp-search-api/main.go

```diff
@@ -50,7 +51,16 @@ func main() {
 		os.Exit(1)
 	}
 
-	elasticSearchClient := elasticsearch.New(cfg.ElasticSearchAPIURL, dphttp.NewClient(), cfg.SignElasticsearchRequests)
+	var esSigner *esauth.Signer
+	if cfg.SignElasticsearchRequests {
+		esSigner, err = esauth.NewAwsSigner("", "", cfg.AwsRegion, cfg.AwsService)
+		if err != nil {
+			log.Event(nil, "failed to create aws v4 signer", log.ERROR, log.Error(err))
+			os.Exit(1)
+		}
+	}
+
+	elasticSearchClient := elasticsearch.New(cfg.ElasticSearchAPIURL, dphttp.NewClient(), cfg.SignElasticsearchRequests, esSigner, cfg.AwsRegion, cfg.AwsService)
```

Comment:
Question:

Just checking if its OK for `esSigner` to be `nil` inside the `elasticseach.New` func? If it is nil is that handled or will it error? I assume its OK but best to double check.

Reply:
Yes, see my message below.